Library/Homebrew/formula_versions.rb

```diff
@@ -61,11 +61,21 @@ def bottle_version_map(branch)
   end
 
   def revision_map(branch)
+    version_attribute_map(:revision, branch)
+  end
+
+  def version_scheme_map(branch)
+    version_attribute_map(:version_scheme, branch)
+  end
```

Comment:
This has quite bad performance. As it's loading formula twice(which it's quite costly).